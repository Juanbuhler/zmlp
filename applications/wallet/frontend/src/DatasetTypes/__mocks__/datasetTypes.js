const datasetTypes = {
  results: [
    {
      name: 'FaceRecognition',
      label: 'Face Recognition',
      description: 'Use when labeling detected faces.',
    },
    {
      name: 'Classification',
      label: 'Classification',
      description: 'Use when classifying the asset.',
    },
    {
      name: 'ObjectDetection',
      label: 'Object Detection',
      description: 'Use when training for objects.',
    },
  ],
}

export default datasetTypes
