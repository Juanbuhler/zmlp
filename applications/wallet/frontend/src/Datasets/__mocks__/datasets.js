const datasets = {
  count: 4,
  next: null,
  previous: null,
  results: [
    {
      id: '111d0534-7736-135f-8a62-d29da837913f',
      projectId: '4b89213b-ce4f-4588-9593-2a20c5a2761f',
      name: 'faces',
      type: 'FaceRecognition',
      description: 'A dataset for model training',
      modelCount: 0,
      timeCreated: 1618511669753,
      timeModified: 1618511669753,
    },
    {
      id: '4b0b10a8-cec1-155c-b12f-ee2bc8787e06',
      projectId: '4b89213b-ce4f-4588-9593-2a20c5a2761f',
      name: 'cats',
      type: 'Classification',
      description: 'A dataset for model training',
      modelCount: 1,
      timeCreated: 1617745633084,
      timeModified: 1617745633084,
    },
    {
      id: '1c237c5c-8b64-11eb-882c-a6e78dab3824',
      projectId: '4b89213b-ce4f-4588-9593-2a20c5a2761f',
      name: 'pets',
      type: 'ObjectDetection',
      description: 'A dataset for model training',
      modelCount: 2,
      timeCreated: 1616454765001,
      timeModified: 1616454765001,
    },
    {
      id: '4b0b10a9-cec1-155c-b12f-ee2bc8787e06',
      projectId: '4b89213b-ce4f-4588-9593-2a20c5a2761f',
      name: 'mystery',
      type: 'UnknownType',
      description: 'A dataset for model training',
      modelCount: 0,
      timeCreated: 1617746190788,
      timeModified: 1617746190788,
    },
  ],
}

export default datasets
