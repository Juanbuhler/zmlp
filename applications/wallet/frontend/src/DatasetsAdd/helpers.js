import Router from 'next/router'

import {
  fetcher,
  revalidate,
  getQueryString,
  parseResponse,
} from '../Fetch/helpers'

export const onSubmit = async ({
  dispatch,
  projectId,
  state: { name, description, type },
}) => {
  dispatch({ isLoading: true, errors: {} })

  try {
    const { id: datasetId } = await fetcher(
      `/api/v1/projects/${projectId}/datasets/`,
      {
        method: 'POST',
        body: JSON.stringify({ name, description, type }),
      },
    )

    await revalidate({
      key: `/api/v1/projects/${projectId}/datasets/`,
    })

    const queryString = getQueryString({
      action: 'add-dataset-success',
      datasetId,
    })

    Router.push(`/[projectId]/datasets${queryString}`, `/${projectId}/datasets`)
  } catch (response) {
    const errors = await parseResponse({ response })

    dispatch({ isLoading: false, errors })
  }
}
