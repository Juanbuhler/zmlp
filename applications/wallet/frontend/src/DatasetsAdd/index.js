import Head from 'next/head'

import PageTitle from '../PageTitle'
import BetaBadge from '../BetaBadge'
import Tabs from '../Tabs'
import SuspenseBoundary, { ROLES } from '../SuspenseBoundary'

import DatasetsAddForm from './Form'

const Datasets = () => {
  return (
    <>
      <Head>
        <title>Datasets</title>
      </Head>

      <PageTitle>
        <BetaBadge isLeft />
        Datasets
      </PageTitle>

      <Tabs
        tabs={[
          { title: 'View all', href: '/[projectId]/datasets' },
          { title: 'Create New Dataset', href: '/[projectId]/datasets/add' },
        ]}
      />

      <SuspenseBoundary role={ROLES.ML_Tools}>
        <DatasetsAddForm />
      </SuspenseBoundary>
    </>
  )
}

export default Datasets
