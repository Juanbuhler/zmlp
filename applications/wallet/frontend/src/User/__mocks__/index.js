import { createContext, createElement } from 'react'

import user from './user'

export const noop = () => {}

export const UserContext = createContext({
  user,
  googleAuth: {},
  setGoogleAuth: noop,
})

const User = ({ children, ...rest }) => createElement('User', rest, children)

export default User
