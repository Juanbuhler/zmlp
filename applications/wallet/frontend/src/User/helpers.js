export const noop = () => {}

export const meFetcher =
  typeof window === 'undefined'
    ? noop
    : async (url) => {
        try {
          const response = await fetch(url)
          if (response.status >= 400) throw response
          return response.json()
        } catch (error) {
          return {}
        }
      }
