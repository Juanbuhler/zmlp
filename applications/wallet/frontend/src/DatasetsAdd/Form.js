import { useReducer } from 'react'
import { useRouter } from 'next/router'
import useSWR from 'swr'

import { spacing } from '../Styles'

import Form from '../Form'
import FlashMessageErrors from '../FlashMessage/Errors'
import SectionTitle from '../SectionTitle'
import SectionSubTitle from '../SectionSubTitle'
import Input, { VARIANTS as INPUT_VARIANTS } from '../Input'
import Textarea, { VARIANTS as TEXTAREA_VARIANTS } from '../Textarea'
import Radio from '../Radio'
import Button, { VARIANTS as BUTTON_VARIANTS } from '../Button'
import ButtonGroup from '../Button/Group'

import { onSubmit } from './helpers'

const INITIAL_STATE = {
  name: '',
  type: '',
  description: '',
  isLoading: false,
  errors: {},
}

const reducer = (state, action) => ({ ...state, ...action })

const DatasetsAddForm = () => {
  const {
    query: { projectId },
  } = useRouter()

  const {
    data: { results: datasetTypes },
  } = useSWR(`/api/v1/projects/${projectId}/datasets/dataset_types/`)

  const [state, dispatch] = useReducer(reducer, INITIAL_STATE)

  return (
    <Form>
      <FlashMessageErrors
        errors={state.errors}
        styles={{ marginTop: -spacing.base, paddingBottom: spacing.comfy }}
      />

      <SectionTitle>Dataset Name &amp; Description</SectionTitle>

      <Input
        autoFocus
        id="name"
        variant={INPUT_VARIANTS.SECONDARY}
        label="Dataset Name:"
        type="text"
        value={state.name}
        onChange={({ target: { value } }) => {
          dispatch({ name: value })
        }}
        hasError={state.errors.name !== undefined}
        errorMessage={state.errors.name}
      />

      <Textarea
        id="description"
        variant={TEXTAREA_VARIANTS.SECONDARY}
        label="Description (optional):"
        value={state.description}
        onChange={({ target: { value } }) => {
          dispatch({ description: value })
        }}
        hasError={state.errors.description !== undefined}
        errorMessage={state.errors.description}
      />

      <SectionTitle>Select a Dataset Type</SectionTitle>

      <SectionSubTitle>
        The type of Dataset determines what kind of Labels to create and which
        Models can be associated with it.
      </SectionSubTitle>

      {datasetTypes.map(({ name, label, description }) => {
        return (
          <div key={name} css={{ paddingTop: spacing.normal }}>
            <Radio
              option={{
                value: name,
                label,
                legend: description,
                initialValue: state.type === name,
              }}
              onClick={({ value }) => dispatch({ type: value })}
            />
          </div>
        )
      })}

      <div css={{ height: spacing.normal }} />

      <ButtonGroup>
        <Button
          type="submit"
          variant={BUTTON_VARIANTS.PRIMARY}
          onClick={() => onSubmit({ projectId, state, dispatch })}
          isDisabled={!state.name || !state.type || state.isLoading}
        >
          {state.isLoading ? 'Creating...' : 'Create New Dataset'}
        </Button>
      </ButtonGroup>
    </Form>
  )
}

export default DatasetsAddForm
