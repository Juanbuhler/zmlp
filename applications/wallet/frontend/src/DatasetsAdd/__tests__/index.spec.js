import TestRenderer, { act } from 'react-test-renderer'

import datasets from '../../Datasets/__mocks__/datasets'
import datasetTypes from '../../DatasetTypes/__mocks__/datasetTypes'
import mockUser from '../../User/__mocks__/user'

import User from '../../User'

import DatasetsAdd from '..'

const noop = () => () => {}

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'
const DATASET_ID = datasets.results[0].id

describe('<DatasetsAdd />', () => {
  it('should render properly', async () => {
    const mockFn = jest.fn()

    require('next/router').__setMockPushFunction(mockFn)

    require('next/router').__setUseRouter({
      pathname: '/[projectId]/datasets/add',
      query: { projectId: PROJECT_ID },
    })

    require('swr').__setMockUseSWRResponse({ data: datasetTypes })

    const component = TestRenderer.create(
      <User initialUser={mockUser}>
        <DatasetsAdd />
      </User>,
    )

    expect(component.toJSON()).toMatchSnapshot()

    // Input valid name
    act(() => {
      component.root
        .findByProps({ id: 'name' })
        .props.onChange({ target: { value: 'My New Dataset' } })
    })

    // Input valid description
    act(() => {
      component.root
        .findByProps({ id: 'description' })
        .props.onChange({ target: { value: 'Lorem Ipsum' } })
    })

    // Select valid type
    act(() => {
      component.root
        .findByProps({ value: 'ObjectDetection' })
        .props.onClick({ value: 'ObjectDetection' })
    })

    expect(component.toJSON()).toMatchSnapshot()

    // Mock Failure
    fetch.mockResponseOnce(JSON.stringify({ name: ['Name already in use'] }), {
      status: 400,
    })

    // Click Submit
    await act(async () => {
      component.root
        .findByProps({ type: 'submit', children: 'Create New Dataset' })
        .props.onClick({ preventDefault: noop })
    })

    // Mock Unknown Failure
    fetch.mockRejectOnce(null, { status: 500 })

    // Click Submit
    await act(async () => {
      component.root
        .findByProps({ type: 'submit', children: 'Create New Dataset' })
        .props.onClick({ preventDefault: noop })
    })

    // Mock Success
    fetch.mockResponseOnce(JSON.stringify({ id: DATASET_ID }), {
      headers: { 'content-type': 'application/json' },
    })

    // Click Submit
    await act(async () => {
      component.root
        .findByProps({ type: 'submit', children: 'Create New Dataset' })
        .props.onClick({ preventDefault: noop })
    })

    expect(fetch.mock.calls.length).toEqual(5)

    expect(fetch.mock.calls[0][0]).toEqual(
      `/api/v1/projects/${PROJECT_ID}/datasets/`,
    )

    expect(fetch.mock.calls[0][1]).toEqual({
      method: 'POST',
      headers: {
        'Content-Type': 'application/json;charset=UTF-8',
        'X-CSRFToken': 'CSRF_TOKEN',
      },
      body: JSON.stringify({
        name: 'My New Dataset',
        description: 'Lorem Ipsum',
        type: 'ObjectDetection',
      }),
    })

    expect(mockFn).toHaveBeenCalledWith(
      `/[projectId]/datasets?action=add-dataset-success&datasetId=${DATASET_ID}`,
      `/${PROJECT_ID}/datasets`,
    )
  })
})
