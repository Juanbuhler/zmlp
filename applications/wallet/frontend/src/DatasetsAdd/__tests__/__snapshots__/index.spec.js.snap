// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatasetsAdd /> should render properly 1`] = `
Array [
  .emotion-0 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-top: 24px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  color: #ffffff;
  background-color: #d6680b;
  border-radius: 32px;
  font-size: 11px;
  line-height: 15px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.2px;
  padding: 1px;
  padding-left: 6px;
  padding-right: 6px;
  margin-right: 8px;
}

<h2
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      beta
    </span>
    Datasets
  </h2>,
  .emotion-0 {
  padding: 16px;
  padding-left: 0;
  padding-right: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-1 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #4a4a4a;
}

.emotion-2 {
  padding-right: 16px;
  margin-bottom: -1px;
}

.emotion-3 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 0;
  color: #b3b3b3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3A5DFF;
}

.emotion-5 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<nav
    className="emotion-0"
  >
    <ul
      className="emotion-1"
    >
      <li
        className="emotion-2"
      >
        <a
          className="emotion-3"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/datasets"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          View all
        </a>
      </li>
      <li
        className="emotion-2"
      >
        <a
          className="emotion-5"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/datasets/add"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Create New Dataset
        </a>
      </li>
    </ul>
  </nav>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 470px;
  padding: 16px;
  padding-left: 0;
  padding-right: 0;
}

.emotion-1 {
  font-size: 21px;
  line-height: 30px;
  font-weight: 500;
  padding-top: 16px;
}

.emotion-2 {
  padding-top: 8px;
}

.emotion-3 {
  display: block;
  padding-bottom: 8px;
  color: #b3b3b3;
}

.emotion-4 {
  position: relative;
}

.emotion-5 {
  outline-offset: 0;
  padding: 10px 12px;
  border-radius: 2px;
  box-shadow: inset 0 1px 3px 0 transparent;
  width: 100%;
  cursor: text;
  color: #ffffff;
  background-color: #3b3b3b;
  border: 1px solid transparent;
  border-width: 2px;
}

.emotion-5:focus,
.emotion-5 .noop {
  border: 2px solid #3A5DFF;
  outline: #3A5DFF;
  color: #000000;
  background-color: #ffffff;
}

.emotion-5:hover,
.emotion-5 .noop {
  border: 2px solid #ffffff;
}

.emotion-6 {
  padding-top: 8px;
}

.emotion-7 {
  padding-top: 8px;
}

.emotion-8 {
  display: block;
  padding-bottom: 8px;
  color: #b3b3b3;
}

.emotion-9 {
  position: relative;
}

.emotion-10 {
  outline-offset: 0;
  padding: 10px 12px;
  border-radius: 2px;
  box-shadow: inset 0 1px 3px 0 transparent;
  width: 100%;
  font-size: 15px;
  line-height: 19px;
  color: #ffffff;
  background-color: #3b3b3b;
  border: 1px solid transparent;
  border-width: 2px;
}

.emotion-10:focus,
.emotion-10 .noop {
  border: 2px solid #3A5DFF;
  background-color: #ffffff;
  outline: #3A5DFF;
  color: #000000;
}

.emotion-13 {
  margin: 0;
  padding-top: 8px;
  color: #b3b3b3;
}

.emotion-14 {
  padding-top: 16px;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  padding-top: 1px;
}

.emotion-17 {
  padding: 0;
  -webkit-appearance: none;
  border-radius: 16px;
  width: 16px;
  height: 16px;
  border: 1px solid #808080;
}

.emotion-18 {
  position: absolute;
  width: 8px;
  height: 8px;
  opacity: 100;
  border-radius: 16px;
  background-color: none;
}

.emotion-19 {
  padding-left: 8px;
}

.emotion-20 {
  font-weight: 700;
  color: #ffffff;
}

.emotion-21 {
  color: #b3b3b3;
}

.emotion-38 {
  height: 16px;
}

.emotion-39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
}

.emotion-39 button,
.emotion-39 a {
  margin-right: 16px;
}

.emotion-39 button:last-child,
.emotion-39 a:last-child {
  margin-right: 0;
}

.emotion-40 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
}

.emotion-40,
.emotion-40:hover,
.emotion-40:visited,
.emotion-40 .noop {
  background-color: #3A5DFF;
}

.emotion-40:hover,
.emotion-40 .noop {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #6088FE;
}

.emotion-40[aria-disabled=true],
.emotion-40 .noop {
  color: #2e2e2e;
  background-color: #808080;
}

<form
    action=""
    className="emotion-0"
    method="post"
    onSubmit={[Function]}
  >
    <h3
      className="emotion-1"
    >
      Dataset Name & Description
    </h3>
    <div
      className="emotion-2"
    >
      <label
        className="emotion-3"
        htmlFor="name"
      >
        Dataset Name:
      </label>
      <div
        className="emotion-4"
      >
        <input
          autoComplete="off"
          autoFocus={true}
          className="emotion-5"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="emotion-6"
      >
         
      </div>
    </div>
    <div
      className="emotion-7"
    >
      <label
        className="emotion-8"
        htmlFor="description"
      >
        Description (optional):
      </label>
      <div
        className="emotion-9"
      >
        <textarea
          className="emotion-10"
          id="description"
          name="description"
          onChange={[Function]}
          rows="5"
          value=""
        />
      </div>
      <div
        className="emotion-7"
      >
         
      </div>
    </div>
    <h3
      className="emotion-1"
    >
      Select a Dataset Type
    </h3>
    <p
      className="emotion-13"
    >
      The type of Dataset determines what kind of Labels to create and which Models can be associated with it.
    </p>
    <div
      className="emotion-14"
    >
      <div>
        <label
          className="emotion-15"
        >
          <div
            className="emotion-16"
          >
            <input
              className="emotion-17"
              defaultChecked={false}
              name="datasetType"
              onClick={[Function]}
              type="radio"
              value="FaceRecognition"
            />
            <div
              className="emotion-18"
            />
          </div>
          <div
            className="emotion-19"
          >
            <div
              className="emotion-20"
            >
              Face Recognition
            </div>
            <div
              className="emotion-21"
            >
              Use when labeling detected faces.
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      className="emotion-14"
    >
      <div>
        <label
          className="emotion-15"
        >
          <div
            className="emotion-16"
          >
            <input
              className="emotion-17"
              defaultChecked={false}
              name="datasetType"
              onClick={[Function]}
              type="radio"
              value="Classification"
            />
            <div
              className="emotion-18"
            />
          </div>
          <div
            className="emotion-19"
          >
            <div
              className="emotion-20"
            >
              Classification
            </div>
            <div
              className="emotion-21"
            >
              Use when classifying the asset.
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      className="emotion-14"
    >
      <div>
        <label
          className="emotion-15"
        >
          <div
            className="emotion-16"
          >
            <input
              className="emotion-17"
              defaultChecked={false}
              name="datasetType"
              onClick={[Function]}
              type="radio"
              value="ObjectDetection"
            />
            <div
              className="emotion-18"
            />
          </div>
          <div
            className="emotion-19"
          >
            <div
              className="emotion-20"
            >
              Object Detection
            </div>
            <div
              className="emotion-21"
            >
              Use when training for objects.
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      className="emotion-38"
    />
    <div
      className="emotion-39"
    >
      <button
        aria-disabled={true}
        className="emotion-40"
        onClick={[Function]}
        type="submit"
      >
        Create New Dataset
      </button>
    </div>
  </form>,
]
`;

exports[`<DatasetsAdd /> should render properly 2`] = `
Array [
  .emotion-0 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-top: 24px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  color: #ffffff;
  background-color: #d6680b;
  border-radius: 32px;
  font-size: 11px;
  line-height: 15px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.2px;
  padding: 1px;
  padding-left: 6px;
  padding-right: 6px;
  margin-right: 8px;
}

<h2
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      beta
    </span>
    Datasets
  </h2>,
  .emotion-0 {
  padding: 16px;
  padding-left: 0;
  padding-right: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-1 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #4a4a4a;
}

.emotion-2 {
  padding-right: 16px;
  margin-bottom: -1px;
}

.emotion-3 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 0;
  color: #b3b3b3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3A5DFF;
}

.emotion-5 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<nav
    className="emotion-0"
  >
    <ul
      className="emotion-1"
    >
      <li
        className="emotion-2"
      >
        <a
          className="emotion-3"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/datasets"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          View all
        </a>
      </li>
      <li
        className="emotion-2"
      >
        <a
          className="emotion-5"
          href="/76917058-b147-4556-987a-0a0f11e46d9b/datasets/add"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Create New Dataset
        </a>
      </li>
    </ul>
  </nav>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 470px;
  padding: 16px;
  padding-left: 0;
  padding-right: 0;
}

.emotion-1 {
  font-size: 21px;
  line-height: 30px;
  font-weight: 500;
  padding-top: 16px;
}

.emotion-2 {
  padding-top: 8px;
}

.emotion-3 {
  display: block;
  padding-bottom: 8px;
  color: #b3b3b3;
}

.emotion-4 {
  position: relative;
}

.emotion-5 {
  outline-offset: 0;
  padding: 10px 12px;
  border-radius: 2px;
  box-shadow: inset 0 1px 3px 0 transparent;
  width: 100%;
  cursor: text;
  color: #ffffff;
  background-color: #3b3b3b;
  border: 1px solid transparent;
  border-width: 2px;
}

.emotion-5:focus,
.emotion-5 .noop {
  border: 2px solid #3A5DFF;
  outline: #3A5DFF;
  color: #000000;
  background-color: #ffffff;
}

.emotion-5:hover,
.emotion-5 .noop {
  border: 2px solid #ffffff;
}

.emotion-6 {
  padding-top: 8px;
}

.emotion-7 {
  padding-top: 8px;
}

.emotion-8 {
  display: block;
  padding-bottom: 8px;
  color: #b3b3b3;
}

.emotion-9 {
  position: relative;
}

.emotion-10 {
  outline-offset: 0;
  padding: 10px 12px;
  border-radius: 2px;
  box-shadow: inset 0 1px 3px 0 transparent;
  width: 100%;
  font-size: 15px;
  line-height: 19px;
  color: #ffffff;
  background-color: #3b3b3b;
  border: 1px solid transparent;
  border-width: 2px;
}

.emotion-10:focus,
.emotion-10 .noop {
  border: 2px solid #3A5DFF;
  background-color: #ffffff;
  outline: #3A5DFF;
  color: #000000;
}

.emotion-13 {
  margin: 0;
  padding-top: 8px;
  color: #b3b3b3;
}

.emotion-14 {
  padding-top: 16px;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  padding-top: 1px;
}

.emotion-17 {
  padding: 0;
  -webkit-appearance: none;
  border-radius: 16px;
  width: 16px;
  height: 16px;
  border: 1px solid #808080;
}

.emotion-18 {
  position: absolute;
  width: 8px;
  height: 8px;
  opacity: 100;
  border-radius: 16px;
  background-color: none;
}

.emotion-19 {
  padding-left: 8px;
}

.emotion-20 {
  font-weight: 700;
  color: #ffffff;
}

.emotion-21 {
  color: #b3b3b3;
}

.emotion-33 {
  padding: 0;
  -webkit-appearance: none;
  border-radius: 16px;
  width: 16px;
  height: 16px;
  border: 1px solid #ffffff;
}

.emotion-34 {
  position: absolute;
  width: 8px;
  height: 8px;
  opacity: 100;
  border-radius: 16px;
  background-color: #3A5DFF;
}

.emotion-38 {
  height: 16px;
}

.emotion-39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
}

.emotion-39 button,
.emotion-39 a {
  margin-right: 16px;
}

.emotion-39 button:last-child,
.emotion-39 a:last-child {
  margin-right: 0;
}

.emotion-40 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-40,
.emotion-40:hover,
.emotion-40:visited,
.emotion-40 .noop {
  background-color: #3A5DFF;
}

.emotion-40:hover,
.emotion-40 .noop {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #6088FE;
}

.emotion-40[aria-disabled=true],
.emotion-40 .noop {
  color: #2e2e2e;
  background-color: #808080;
}

<form
    action=""
    className="emotion-0"
    method="post"
    onSubmit={[Function]}
  >
    <h3
      className="emotion-1"
    >
      Dataset Name & Description
    </h3>
    <div
      className="emotion-2"
    >
      <label
        className="emotion-3"
        htmlFor="name"
      >
        Dataset Name:
      </label>
      <div
        className="emotion-4"
      >
        <input
          autoComplete="off"
          autoFocus={true}
          className="emotion-5"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value="My New Dataset"
        />
      </div>
      <div
        className="emotion-6"
      >
         
      </div>
    </div>
    <div
      className="emotion-7"
    >
      <label
        className="emotion-8"
        htmlFor="description"
      >
        Description (optional):
      </label>
      <div
        className="emotion-9"
      >
        <textarea
          className="emotion-10"
          id="description"
          name="description"
          onChange={[Function]}
          rows="5"
          value="Lorem Ipsum"
        />
      </div>
      <div
        className="emotion-7"
      >
         
      </div>
    </div>
    <h3
      className="emotion-1"
    >
      Select a Dataset Type
    </h3>
    <p
      className="emotion-13"
    >
      The type of Dataset determines what kind of Labels to create and which Models can be associated with it.
    </p>
    <div
      className="emotion-14"
    >
      <div>
        <label
          className="emotion-15"
        >
          <div
            className="emotion-16"
          >
            <input
              className="emotion-17"
              defaultChecked={false}
              name="datasetType"
              onClick={[Function]}
              type="radio"
              value="FaceRecognition"
            />
            <div
              className="emotion-18"
            />
          </div>
          <div
            className="emotion-19"
          >
            <div
              className="emotion-20"
            >
              Face Recognition
            </div>
            <div
              className="emotion-21"
            >
              Use when labeling detected faces.
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      className="emotion-14"
    >
      <div>
        <label
          className="emotion-15"
        >
          <div
            className="emotion-16"
          >
            <input
              className="emotion-17"
              defaultChecked={false}
              name="datasetType"
              onClick={[Function]}
              type="radio"
              value="Classification"
            />
            <div
              className="emotion-18"
            />
          </div>
          <div
            className="emotion-19"
          >
            <div
              className="emotion-20"
            >
              Classification
            </div>
            <div
              className="emotion-21"
            >
              Use when classifying the asset.
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      className="emotion-14"
    >
      <div>
        <label
          className="emotion-15"
        >
          <div
            className="emotion-16"
          >
            <input
              className="emotion-33"
              defaultChecked={true}
              name="datasetType"
              onClick={[Function]}
              type="radio"
              value="ObjectDetection"
            />
            <div
              className="emotion-34"
            />
          </div>
          <div
            className="emotion-19"
          >
            <div
              className="emotion-20"
            >
              Object Detection
            </div>
            <div
              className="emotion-21"
            >
              Use when training for objects.
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      className="emotion-38"
    />
    <div
      className="emotion-39"
    >
      <button
        className="emotion-40"
        onClick={[Function]}
        type="submit"
      >
        Create New Dataset
      </button>
    </div>
  </form>,
]
`;
